compositor_render/src/transformations/shader/error.rs

```diff
@@ -147,3 +143,37 @@ pub enum ParametersValidationError {
         error: Box<ParametersValidationError>,
     },
 }
+
+pub(crate) trait ShaderGlobalVariableExt {
+    fn to_string(&self) -> String;
+}
+
+impl ShaderGlobalVariableExt for naga::GlobalVariable {
+    fn to_string(&self) -> String {
+        let group_and_binding = match self.binding {
+            Some(naga::ResourceBinding { group, binding }) => {
+                format!("@group({}) @binding({}) ", group, binding)
+            }
+            None => "".to_string(),
+        };
+        let space = match self.space {
+            naga::AddressSpace::Function => "<function>".to_string(),
+            naga::AddressSpace::Private => "<private>".to_string(),
+            naga::AddressSpace::WorkGroup => "<workgroup>".to_string(),
+            naga::AddressSpace::Uniform => "<uniform>".to_string(),
+            naga::AddressSpace::Storage { access } => {
+                let access = match access.bits() {
+                    1 => "read",
+                    2 => "write",
+                    3 => "read_write",
+                    _ => "",
+                };
+                format!("<storage, {}>", access)
+            }
+            naga::AddressSpace::Handle => "".to_string(),
```

Comment:
Not sure what it means, but I got this error when missing "<uniform>" for the user-provided params (binding 1). Make sure this type of error is informative